File: src/easy/sort-array-by-parity-two.ts
export const sortArrayByParityTwo = (nums: number[]): number[] => {
    const result = Array(nums.length);
    let even = 0;
    let odd = 0;
    for (const num of nums) {
        if (num % 2 === 0) {
            result[even] = num;
            even += 2;
        } else {
            result[odd] = num;
            odd += 2;
        }
    }
    return result;
};

File: src/easy/sort_array_by_parity_two.rs
pub fn sort_array_by_parity_two(nums: Vec<i32>) -> Vec<i32> {
    let mut result = vec![0; nums.len()];
    let mut even: usize = 0;
    let mut odd: usize = 1;
    for &num in nums.iter() {
        if num % 2 == 0 {
            result[even] = num;
            even += 2;
        } else {
            result[odd] = num;
            odd += 2;
        }
    }
    result
}
